src/did/src/transaction.rs

```diff
@@ -85,6 +91,68 @@ impl From<u64> for BlockNumber {
     }
 }
 
+#[derive(Debug, Display, Clone, PartialEq, Eq, From)]
+pub enum BlockId {
+    BlockNumber(BlockNumber),
+    BlockHash(H256),
+}
+
+impl Serialize for BlockId {
```

Comment:
Why not derive Serialize, Deserialize and CandidType for the BlockId?

Reply:
Why do we need the flattened representation?

Reply:
@ufoscout Because it corresponds to how the value is represented as a JSON RPC parameter: it is either a hash string, a hex number string or a tag ("latest", "pending", etc)